state/block_test.go

```diff
@@ -0,0 +1,75 @@
+package state
+
+import (
+	"fmt"
+	"math/big"
+	"os"
+	"reflect"
+	"testing"
+
+	"github.com/ChainSafe/gossamer/common"
+	"github.com/ChainSafe/gossamer/core/types"
+)
+
+func TestGetBlockByNumber(t *testing.T) {
+	dataDir := "../test_data/block"
+
+	//Create a new blockState
+	blockState, err := NewBlockState(dataDir)
+	if err != nil {
+		t.Fatal(err)
+	}
+	//Close DB & erase data dir contents
+	defer func() {
+		err = blockState.Db.Db.Close()
+		if err != nil {
+			t.Fatal("BlockDB close err: ", err)
+		}
+		if err = os.RemoveAll(dataDir); err != nil {
+			fmt.Println("removal of temp directory test_data failed")
```

Comment:
should we `t.Fatal(err)` here rather than print?

Reply:
this could be removed completely and use ioutils intead, see

Reply:
updated to `t.Fatal(err)`, let's address the latter in #503